### ISSUE TITLE
Assumption checks and Spherity corrections in Bayesian ANOVAs

### ISSUE THREAD

**Issue by AugustOlsson** [jasp-stats/jasp-issues#773]
I have two proposals for enhancements.

1). The first one is to include assumption checks in the Bayesian versions of all analyses. This is just a very minor issue, but it should be very easy to just duplicate the buttons from the frequentist menue, and it will save time not to have to go back and forth between the frequentist analyses and the bayesian ones just to check for assumptions.

2). Is it possible to include bayesian versions of e.g. the Greenhouse-Geisser correction when spherity checks fail for repeated measure BANOVAs? I am unsure how to approach a bayesian repeated measure BANOVA when spherity checks fails atm, because there are no bayesian corrections in JASP and I would guess that the Bayesian ANOVA will be affected in a similar manner as a the frequentist one when e.g. the assumption of spherity is not fullfulled?

Best,
August

**Comment by boutinb** [comment]
@JohnnyDoorn could you answer this request?

**Comment by JohnnyDoorn** [comment]
Hi @AugustOlsson,

For the assumption of normality, you can use the Q-Q plot option. This plot is done a little bit differently than for the frequentist version - this is an excerpt of [Don van den Bergh's tutorial on Bayesian ANOVA](https://psyarxiv.com/spreb/):

> In contrast to a frequentist ANOVA, where the residuals are point estimates, a Bayesian
ANOVA provides a probability distribution for each residual. The uncertainty in the residuals can thus be summarized by 95% credible intervals. 

Unfortunately, the sphericity corrections and other assumption checks are not developed yet for the Bayesian ANOVA (you can read a short discussion [here](https://forum.cogsci.nl/discussion/4073/testing-assumptions-for-bayesian-repeated-measure-anova). We are currently setting up a research project that aims to investigate best practices in Bayesian ANOVA, including violations of assumptions. 

Kind regards,
Johnny

**Comment by AugustOlsson** [comment]
Hej Johnny,

Thanks for the quick reply!
Great to hear that you are working on Sphericity corrections! 

I should have been more clear about assumptions checks, my apologies. I was mainly thinking about the assumption of homogeneous variance. I understand that you would need to put a lot of time into developing a bayesian version of e.g. Levene's test, but shoudn't the frequentist version of Levene's test be valid for the bayesan ANOVA as well? If so, could you then include the option to perform Levene's (frequentist) test for homogenous varriance in the Bayesian ANOVA-module? Then one would not have to jump back and forth between a frequentist and bayesian ANOVA to check the homogenity of varriance assumption.

Best,
August

**Comment by EJWagenmakers** [comment]
But that would pollute our Bayesian method. I prefer waiting just a little while longer until the Bayesian test is implemented.

**Comment by AugustOlsson** [comment]
Yeah I understand if you want to keep the Bayesian module clear of frequentist test. In the meanwhile one can of course perform frequentist checks elsewhere. 

Best,
August

**Comment by tomtomme** [comment]
@AugustOlsson 
Just as a short update. I did have a look for a bayesian version, but this seems still research area. There is however a recent papter that suggests, that sphericity violation "biases can be avoided or at least reduced by fitting the Bayesian models to nonaggregated data and estimating the full random effects structure."

Schad, D. J., Nicenboim, B., & Vasishth, S. (2024). Data aggregation can lead to biased inferences in Bayesian linear mixed models and Bayesian analysis of variance. Psychological Methods. Advance online publication. [https://doi.org/10.1037/met0000621](https://psycnet.apa.org/doi/10.1037/met0000621)

**Comment by AugustOlsson** [comment]
Cool, thanks for the update @tomtomme
